src/c_api/factorization/cholesky.h

```diff
@@ -0,0 +1,121 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2023, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#include <blas.hh>
+#include <blas/util.hh>
+#include <mpi.h>
+
+#include <pika/init.hpp>
+
+#include <dlaf/factorization/cholesky.h>
+#include <dlaf/matrix/matrix.h>
+#include <dlaf/matrix/matrix_mirror.h>
+#include <dlaf_c/desc.h>
+#include <dlaf_c/grid.h>
+
+#include "../blacs.h"
+#include "../grid.h"
+
+template <typename T>
+int cholesky(int dlaf_context, char uplo, T* a, DLAF_descriptor dlaf_desca) noexcept {
+  try {
+    using MatrixMirror = dlaf::matrix::MatrixMirror<T, dlaf::Device::Default, dlaf::Device::CPU>;
+
+    DLAF_ASSERT(dlaf_desca.i == 0, dlaf_desca.i);
+    DLAF_ASSERT(dlaf_desca.j == 0, dlaf_desca.j);
+
+    pika::resume();
+
+    auto communicator_grid = dlaf_grids.at(dlaf_context);
+
+    dlaf::GlobalElementSize matrix_size(dlaf_desca.m, dlaf_desca.n);
+    dlaf::TileElementSize block_size(dlaf_desca.mb, dlaf_desca.nb);
+
+    dlaf::comm::Index2D src_rank_index(dlaf_desca.isrc, dlaf_desca.jsrc);
+
+    dlaf::matrix::Distribution distribution(matrix_size, block_size, communicator_grid.size(),
+                                            communicator_grid.rank(), src_rank_index);
+
+    dlaf::matrix::LayoutInfo layout = colMajorLayout(distribution, dlaf_desca.ld);
+
+    dlaf::matrix::Matrix<T, dlaf::Device::CPU> matrix_host(std::move(distribution), layout, a);
```

Comment:
I would consider to include this code in a function such that it can be reused in all the c wrappers.